### ISSUE TITLE
TypeError: unhashable type: 'list' with pydantic version <2.5.2

### ISSUE THREAD

**Issue by mattibo** [yanyongyu/githubkit#58]
when i run this code i get an TypeError after i upgraded to pydantic version 2.5:

```python
from githubkit import GitHub

with GitHub() as github:
    resp = github.rest.repos.get(owner="yanyongyu", repo="githubkit")
    repo = resp.parsed_data
```

following error is raised:

```console
Traceback (most recent call last):
  File "githubkit\test.py", line 5, in <module>
    repo = resp.parsed_data
           ^^^^^^^^^^^^^^^^
  File "githubkit\githubkit\response.py", line 50, in parsed_data
    return TypeAdapter(self._data_model).validate_json(self.content)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "githubkit\venv\Lib\site-packages\pydantic\type_adapter.py", line 256, in validate_json
    return self.validator.validate_json(__data, strict=strict, context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'list'
```

maybe this has something todo with the new json parsing library pydantic hast introduced with version 2.5: https://docs.pydantic.dev/2.5/concepts/json/#json-parsing

**Comment by yanyongyu** [comment]
it seems the following code also reports the same error:

```python
TypeAdapter(resp._data_model).validate_python(resp.json())
```

Is this a bug of pydantic?

**Comment by mattibo** [comment]
The bug is fixed with version 2.5.2 of pydantic: https://github.com/pydantic/pydantic/releases/tag/v2.5.2

Maybe ignore the version 2.5.0 and 2.5.1 in `pyproject.toml` of githubkit?
